### ISSUE TITLE
Snap applications look wrong.

### ISSUE THREAD

**Issue by philberthfz** [grassmunk/Chicago95#293]
I have a partial fix, but you probably aren't going to like it.

At present, snap applications are unable to access any themes which are not also installed as snaps. This is unfortunate since several users (such as myself) get Xfce through Ubuntu, and if they just install apps from the software store then several of them will not be properly themed. 

Through some very incredibly frustrating testing, I have sketched out an exceedingly limited barebones snapcraft.yaml file which, at least on my system, generates a snap package that will if a command is run to manually connect it to your other snaps, allow your snaps to look mostly correct if you're using the default chicago95 theme. Plus support is not available, and given the way snaps work I'm not sure it'd ever be possible, short of having the plus application generate, install, and connect a new snap for every theme it converts (which would definitely frustrate end-users who actually use snaps).

For reasons which are not clear to me, on one of the machines I've tested, the icon theme isn't being applied correctly resulting in slightly incorrect window glyphs and mis-matched icons. If this situation were resolved, I'd probably recommend considering submitting the snap to the snap store for publication, but as is it's just slightly too jank.

Anyhow, in theory, if one placed the snapcraft.yaml from the attached zip file in the same directory as installer.py and then ran `snapcraft` from that location, it would generate a not-signed snap file. Since the snap is not signed, you would install it using the `--dangerous` switch (e.g: `snap install ./chicago95_3.0.1_all.snap --dangerous`). Once the generated snap is installed, the user would have to run `for i in $(snap connections | grep gtk-common-themes:gtk-3-themes | awk '{print $2}'); do sudo snap connect $i chicago95:gtk-3-themes; done` to connect their snaps to the theme as well as ` for i in $(snap connections | grep gtk-common-themes:icon-themes | awk '{print $2}'); do sudo snap connect $i chicago95:icon-themes; done` to get the icons and window glyphs working. This would have to be done every time they install a new snap. This is not convenient; it is, however, less annoying than having several apps that look different. 

I am led to believe there either is or will eventually be a way to grant auto-connect behavior to themes for snap packages. I do not know what it is or how to use it, ask me no further questions on the matter.

((For convenience sake, I've also included a pre-compiled snap. I recommend doing the work yourself, but if you don't intend on setting up a snap build environment you can skip straight to installing and testing)).

[snap_stuff.zip](https://github.com/grassmunk/Chicago95/files/11896682/snap_stuff.zip)

### tl;dr

Snaps suck for theming - it's doable, but doing it sucks.
I've built a very minimal snap to fill the gap.
Significant manual effort is required to use it because I haven't uploaded it to the snap store.
Plus theming is not supported, if you've enabled a plus theme then snap apps will not be themed and the snap default theme is used instead (adiwata, I think?).
